### PR TITLE
Fix Cython 0.28 build break of qhull.pyx

### LINKED ISSUE #8562: CI: Appveyor builds fail because it can't import ConvexHull from spatial.qhull
The PRs are all poisoned with the same failing 87 tests. I've built locally on a few Windows machines and can't replicate it. 

Is it possible that the cache we use to share between different architectures in Appveyor got poisoned ?

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
Fixes build break reported in https://github.com/cython/cython/issues/2154

Fixes #8562

### CODE REVIEW

**Comment by pv** [comment]
This presumably fails on earlier Cython versions?

**Comment by oleksandr-pavlyk** [comment]
Surprisingly no. Cythonization of modified `qhull.pyx` goes through for me with Cython 0.27.3.

**Comment by ilayn** [comment]
It does fail on Travis though with `__doc__` not writable error.

**Comment by pv** [comment]
If so,
```
try:
    foo.__doc__ = bar.__doc__
except AttributeError:
    foo.__func__.__doc__ = bar.__func__.__doc__
```
probably works on all Cython versions.

**Comment by pv** [comment]
Can you please also revert gh-8566 in this same PR so that we can see
that it actually works.

File: scipy/spatial/qhull.pyx [hunk 1]
```diff
@@ -2353,7 +2362,7 @@ class ConvexHull(_QhullUser):
             self._vertices = np.unique(self.simplices)
         return self._vertices
 
-ConvexHull.add_points.__func__.__doc__ = _QhullUser._add_points.__doc__
+_set_docs(ConvexHull.add_points, _QhullUser._add_points.__doc__)
```
Comment (pv) [inline scipy/spatial/qhull.pyx]: The second argument should not have `.__doc__`.

File: scipy/spatial/qhull.pyx [hunk 2]
```diff
@@ -2158,7 +2158,16 @@ def tsearch(tri, xi):
     """
     return tri.find_simplex(xi)
 
-Delaunay.add_points.__func__.__doc__ = _QhullUser._add_points.__doc__
+# Set docstring for foo to docstring of bar, working around change in Cython 0.28
+# See https://github.com/scipy/scipy/pull/8581
+def _set_docs(foo, bar):
```
Comment (pv) [inline scipy/spatial/qhull.pyx]: `dst, src`, not `foo, bar` probably

**Comment by oleksandr-pavlyk** [comment]
All builds were green. I squashed all changes of `qhull.pyx` into a single commit. Ready to merge.

**Comment by rgommers** [comment]
LGTM. This looks like a safe fix - I'd like to take it along for 1.0.1; an incompatibility with latest Cython is annoying. @pv sound okay?

**Comment by pv** [comment]
Should be safe afaik

**Comment by rgommers** [comment]
Merged. Thanks @oleksandr-pavlyk, @pv!

**Pull request merged**